src/lib/tree/data-source/nested-data-source.ts

```diff
@@ -18,13 +19,24 @@ import {map} from 'rxjs/operators';
  * or collapse. The expansion/collapsion will be handled by TreeControl and each non-leaf node.
  */
 export class MatTreeNestedDataSource<T> extends DataSource<T> {
-  _data = new BehaviorSubject<T[]>([]);
+  _data: BehaviorSubject<T[]>;
 
   /**
    * Data for the nested tree
    */
   get data() { return this._data.value; }
-  set data(value: T[]) { this._data.next(value); }
+  set data(value: T[]) {
+    this._data.next(value);
+
+    if (this.treeControl) {
+      this.treeControl.dataNodes = value;
+    }
+  }
+
+  constructor(private treeControl?: NestedTreeControl<T>, initialData: T[] = []) {
```

Comment:
@jelbourn I'm not sure whether we should make this required at some point. Thoughts?

Reply:
The original design was that the tree control and the data source are decoupled. I'll need to think about this one a bit; I'm generally hesitant to introduce new APIs like this

Reply:
That's reasonable. I'm just thinking that we could leave it as an opt-in API because the `FlatDataSource´ also accepts a `TreeControl` and handles the data propagation.

So, this would be more an alignment with the other data source because I think letting the user sync the data manually is not very convenient. We are also good by just having the error messages.